src/com/google/enterprise/adaptor/opentext/OpentextAdaptor.java

```diff
@@ -675,6 +712,236 @@ GroupPrincipal getGroupPrincipal(Member group) {
+              this.lastModDocId = nodeId;
+              this.lastModDate = new SimpleDateFormat("yyyyMMdd").format(date);
+              this.lastModTime = new SimpleDateFormat("HHmmss").format(date);
+            }
+          }
+        }
+      }
+    } while (resultCount > 0);
+    log.log(Level.FINER, "Sending modified doc ids: {0}", docIds);
+    pusher.pushDocIds(docIds);
+  }
+
+  @VisibleForTesting
+  String getXmlSearchDocId(Element resultElement) {
+    List<String> ids = getXmlSearchIds(resultElement);
+    if (ids.size() == 0) {
+      return null;
+    }
+
+    // OTLocation includes the search result item's
+    // id. LocationPathString doesn't include the item, so remove
+    // the item id here before building the path.
+    String objectId = ids.remove(ids.size() - 1);
+
+    // If the result item is a start point, return its
+    // id. Otherwise, find the index of a start point in the id
+    // path.
+    StartPoint startPoint = null;
+    int startPointIndex = -1;
+    for (StartPoint sp : this.startPoints) {
+      String startPointId = String.valueOf(sp.getNodeId());
+      if (startPointId.equals(objectId)) {
+        return escapeParam(sp.getName()) + ":" + startPointId;
+      } else {
+        startPointIndex = ids.indexOf(startPointId);
+        if (startPointIndex > -1) {
+          startPoint = sp;
+          break;
+        }
+      }
+    }
+    if (startPointIndex == -1) {
+      log.log(Level.FINE, "No start point in path: " + ids);
+      return null;
+    }
+
+    List<String> names = getXmlSearchNames(resultElement);
+    if (names.size() != ids.size()) {
+      log.log(Level.FINE, "Mismatch in names/ids in search results");
+      return null;
+    }
+
+    StringBuilder docId = new StringBuilder();
+    docId.append(escapeParam(startPoint.getName()));
+    for (int i = startPointIndex + 1; i < names.size(); i++) {
+      docId.append("/").append(escapeParam(names.get(i)));
+    }
+    docId.append("/")
+        .append(paramEscaper.escape(getTextContent(resultElement, "OTName")))
+        .append(":")
+        .append(objectId);
+
+    return docId.toString();
+  }
+
+  @VisibleForTesting
+  List<String> getXmlSearchNames(Element resultElement) {
+    List<String> names =
+        Splitter.on(":").trimResults().omitEmptyStrings()
+        .splitToList(getTextContent(resultElement, "LocationPathString"));
+    return names;
+  }
+
+  /* Returns a modifiable list. */
+  @VisibleForTesting
+  List<String> getXmlSearchIds(Element resultElement) {
+    String idString = getTextContent(resultElement, "OTLocation");
+    List<String> ids = new ArrayList<>(
+        Splitter.on(" ").omitEmptyStrings().trimResults()
+        .splitToList(idString));
+    ListIterator<String> iterator = ids.listIterator();
+    while (iterator.hasNext()) {
+      Long idn = Longs.tryParse(iterator.next());
+      if (idn == null) {
+        iterator.remove(); // Keep going, or return?
```

Comment:
Let's just do whatever is necessary to ensure that this value (or the whole idString) is logged and that this search result is skipped over. We may need to account for the checkpoint when this is the last search result. My first thought is not to try retrieving data for this node, but whatever is easiest is probably OK.